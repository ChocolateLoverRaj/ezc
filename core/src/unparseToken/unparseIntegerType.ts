import CoreTokenDatas from '../CoreTokenDatas'
import CoreTokenType from '../CoreTokenType'
import UnparseToken from './UnparseToken'

const unparseIntegerType: UnparseToken<CoreTokenDatas[CoreTokenType.INTEGER_TYPE]> = (
  number
) => `i${number}`

export default unparseIntegerType
