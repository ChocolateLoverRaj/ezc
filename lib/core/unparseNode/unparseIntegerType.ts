import CoreTokenType from '../CoreTokenType'
import CoreNodeDatas from '../parseNode/CoreNodeDatas'
import CoreNodeType from '../parseNode/CoreNodeType'
import UnparsedPartType from './UnparsedPartType'
import UnparseNode from './UnparseNode'

const unparseIntegerType: UnparseNode<CoreNodeDatas[CoreNodeType.INTEGER_TYPE]> = data => [{
  type: UnparsedPartType.TOKEN,
  data: {
    type: {
      enum: CoreTokenType,
      id: CoreTokenType.INTEGER_TYPE
    },
    data
  }
}]

export default unparseIntegerType
