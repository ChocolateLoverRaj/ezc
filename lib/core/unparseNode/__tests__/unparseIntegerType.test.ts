import unparseIntegerType from '../unparseIntegerType'

test('i8', () => {
  expect(unparseIntegerType(8)).toMatchSnapshot()
})
