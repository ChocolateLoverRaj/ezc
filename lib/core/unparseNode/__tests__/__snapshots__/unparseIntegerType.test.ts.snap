// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`i8 1`] = `
Array [
  Object {
    "data": Object {
      "data": 8,
      "type": Object {
        "enum": Object {
          "0": "IDENTIFIER",
          "1": "NUMBER_LITERAL",
          "2": "INTEGER_TYPE",
          "3": "STRING_LITERAL",
          "4": "KEY_WORD",
          "IDENTIFIER": 0,
          "INTEGER_TYPE": 2,
          "KEY_WORD": 4,
          "NUMBER_LITERAL": 1,
          "STRING_LITERAL": 3,
        },
        "id": 2,
      },
    },
    "type": 0,
  },
]
`;
